web-common/src/features/metrics-views/ai-generation/generateMetricsView.ts

```diff
@@ -118,24 +123,64 @@ export function useCreateMetricsViewFromTableUIAction(
           database: database,
           databaseSchema: databaseSchema,
           table: tableName,
-          path: newFilePath,
+          path: newMetricsViewFilePath,
           useAi: false,
         });
       }
 
-      // Preview
       const previousScreenName = getScreenNameFromPage();
-      await goto(`/files${newFilePath}`);
-      void behaviourEvent.fireNavigationEvent(
-        newDashboardName,
-        behaviourEventMedium,
-        metricsEventSpace,
-        previousScreenName,
-        MetricsEventScreenName.Dashboard,
-      );
+
+      if (createExplore) {
```

Comment:
Can we separate this part? This function is doing too much.

Reply:
Hmm, I agree it's a long function. A couple things:
- We shouldn't put `goto` in another function layer.
- I'm nervous to wrap another layer around `handleEntityCreate`, which itself wraps the actual file creation.

I'd propose that, in general, we need to standardize our various resource-to-resource creation functions. I think we should take this case as part of a broader refactor of `handleEntityCreate` that we can more succinctly use everywhere. This is good motivation.